kafka/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KafkaSink.scala

```diff
@@ -88,6 +90,7 @@ object KafkaSink {
       props.setProperty("linger.ms", bufferConfig.timeLimit.toString)
       props.setProperty("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
       props.setProperty("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+      props.setProperty("sasl.login.callback.handler.class", authCallbackClass)
```

Comment:
By setting to the property here, I think you are forcing **all** kafka users (including open-source users) to use Azure authentication.  Don't we anticipate some users want to integrate with non-azure kafka?

The previous implementation set the handler class in the `application.conf` file, so there was a standard way for anyone to override the setting with something different.  Is there a reason that configuration approach won't work anymore?

Reply:
It can still be overwritten actually since producer properties from config is set later on. 

I made it like this because this property is a bit special one. It have to be a class name from the application. Adding this property to `application.conf` looks a bit unsafe because in that case, there won't be any direct reference to callback handlers. Someone later on might remove those classes and application would compile without any problem but this feature would be broken. However, with current implementation, we directly reference the callback handler classes and if they removed from the project, the project wouldn't even compile. Of course, you can say that we might add some test cases to make sure they aren't removed but this implementation looks better to me. 

I don't have strong opinion about this one, I can change it the way you described as well if you want.

Reply:
> It can still be overwritten actually since producer properties from config is set later on.

Ah, my mistake... I mis-read the code earlier.  You are right, it can be overridden.

I'm happy then.